src/components/shadow.js

```diff
@@ -0,0 +1,43 @@
+var component = require('../core/component');
+var THREE = require('../lib/three');
+var bind = require('../utils/bind');
+var registerComponent = component.registerComponent;
+
+/**
+ * Shadow component.
+ *
+ * When applied to an entity, that entity's geometry and any descendants will cast or receive
+ * shadows as specified by the `cast` and `receive` properties.
+ */
+module.exports.Component = registerComponent('shadow', {
+  schema: {
+    cast: {default: true},
+    receive: {default: true}
+  },
+
+  init: function () {
+    this.onMeshChanged = bind(this.update, this);
+    this.el.addEventListener('model-loaded', this.onMeshChanged);
+  },
+
+  update: function () {
+    var data = this.data;
+    this.el.object3D.traverse(function (node) {
+      if (node instanceof THREE.Mesh) {
+        node.castShadow = data.cast;
+        node.receiveShadow = data.receive;
```

Comment:
I think we need to update material here and in remove() like this
to handle set/removeAttribute('shadow') executed after scene is rendered.

    var material = node.material;
    if (material.isMultlMaterial === true) {
        for (var i = 0, il = material.materials.length; i < il; i++) {
            material.materials[i].needsUpdate = true;
        }
    } else {
        material.needsUpdate = true;
    }

Reference:
Run the following commands one by one in the console at
https://threejs.org/examples/webgl_animation_skinning_morph.html

    mesh.receiveShadow = false
    mesh.material.materials[0].needsUpdate = true;
    mesh.receiveShadow = true
    mesh.material.materials[0].needsUpdate = true;